Test/Optimizer/test_sr.py

```diff
@@ -18,36 +18,25 @@ def test_svd_threshold():
     """
     Test SVD threshold option of BDCSVD
     """
-    with pytest.raises(
-        ValueError, match="svd_threshold option only available for BDCSVD solver"
-    ):
-        SR(svd_threshold=1e-3)
-    with pytest.raises(
-        ValueError, match="svd_threshold option only available for BDCSVD solver"
-    ):
+    with pytest.raises(ValueError):
         SR(use_iterative=True, svd_threshold=1e-3)
-    with pytest.raises(
-        ValueError, match="svd_threshold option only available for BDCSVD solver"
-    ):
-        SR(lsq_solver="LDLT", svd_threshold=1e-3)
 
     a = np.diag([1e0 + 0j, 1e-3, 1e-6])
     b = np.array([1.0 + 0j, 1.0, 1.0])
 
     def SR_with_threshold(t):
-        return SR(lsq_solver="BDCSVD", svd_threshold=t, diag_shift=0)
+        return SR(lsq_solver="SVD", svd_threshold=t, diag_shift=0)
 
     def solve(sr, a, b):
         a1 = np.sqrt(a) * np.sqrt(a.shape[0])
-        out = np.copy(b)
-        sr.compute_update(a1, b, out)
+        out = sr.compute_update(a1, b)
         return out
 
     sr = SR_with_threshold(1e-1)
     assert np.allclose(solve(sr, a, b), [1.0, 0.0, 0.0])
 
-    sr = SR_with_threshold(1e-3)
+    sr = SR_with_threshold(1e-4)
```

Comment:
Is this change due to a different interpretation of the `threshold` parameter in the new code?

Reply:
if you are **exactly** at the threshold you basically get random behavior you should be slightly below it to make sure it is working properly

Reply:
Indeed in the old case you already had 
```python
    sr = SR_with_threshold(1e-1)
    assert np.allclose(solve(sr, a, b), [1.0, 0.0, 0.0])
```

i.e. the threshold was one order of magnitude smaller